plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -944,20 +944,27 @@ public function is_valid_for_product( $product, $values = array() ) {
 	 * Converts one of the WC_Coupon message/error codes to a message string and.
 	 * displays the message/error.
 	 *
-	 * @param int $msg_code Message/error code.
+	 * @param int    $msg_code Message/error code.
+	 * @param string $notice_type Notice type.
 	 */
-	public function add_coupon_message( $msg_code ) {
-		$msg = $msg_code < 200 ? $this->get_coupon_error( $msg_code ) : $this->get_coupon_message( $msg_code );
+	public function add_coupon_message( $msg_code, $notice_type = 'success' ) {
+		if ( $msg_code < 200 ) {
+			$msg         = $this->get_coupon_error( $msg_code );
+			$notice_type = 'error';
+		} else {
+			$msg = $this->get_coupon_message( $msg_code );
+		}
```

Comment:
What are your thoughts on simplifying this part as follows, to avoid the redundency:

```suggestion
		if ( $msg_code < 200 ) {
			$notice_type = 'error';
		}
		
		$msg = $this->get_coupon_message( $msg_code );
```

Reply:
I'm afraid we can't do that, as we need to run different methods `get_coupon_error` vs `get_coupon_message`

Reply:
Ah, my bad. I thought we're calling `get_coupon_message()` in both conditions. Making this one as resolved now.